hrabsence/CRM/HRAbsence/Form/AbsenceType.php

```diff
@@ -106,11 +106,32 @@ public function buildQuickForm() {
     $this->addFormRule(array('CRM_HRAbsence_Form_AbsenceType', 'formRule'), $this);
   }
 
-  static function formRule($fields, $files, $self) {
-    $errors = array();
+  /**
+   * Used in validation from addFormRule
+   *
+   * @param array $fields
+   *  An array of fields from the form
+   * @param array $files
+   *  An array of uploaded files
+   * @param self $self
+   *  An instance of this class
+   * @return array
+   *   An array of errors
+   */
+  public static function formRule($fields, $files, $self) {
+    $errors = [];
     if (!array_key_exists('allow_debits', $fields) && !array_key_exists('allow_credits', $fields)) {
       $errors['allow_debits'] = $errors['allow_credits'] = ts("Please choose either 'Allow Debits' and/or 'Allow Credits'");
     }
+
+    $title = CRM_Utils_Array::value('title', $fields);
+    if ($self->getAction() === CRM_Core_Action::ADD) {
+      $existing = civicrm_api3('HRAbsenceType', 'get', ['name' => $title]);
```

Comment:
The problem is that there's no entity validation in Civi. So, if the update comes from a form, we can use the form rules to check things, but when it comes from the API (which is the case when you edit the name directly on the list) that validation will never be called.
There's no out-of-the box solution for this in Civi, unfortunately. What we've been doing in L&A is [this](https://github.com/civicrm/civihr/blob/la-milestone-3/uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/AbsenceType.php#L207-L226). We call this validateParams [inside the create](https://github.com/civicrm/civihr/blob/la-milestone-3/uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/AbsenceType.php#L67) method and then we can be sure the validation will be called either if you use the API or if you call the BAO method directly (which is usually what happens on a form).

This approach has its own problems though:
- Since it throws exceptions, it will stop on the first error instead of doing all the validations and showing all the error messages just once
- The error messages cannot be linked to a form field
- In case of an error, you'll not be redirected to the form to correct whatever is wrong. Instead, you 'll be back to the list, lose what you've entered into the form and just see an error message
- Depending on how the validation methods are implemented, you might lose the ability to update a single field of the entity (what happens when you edit the name from the list)

Anyway, since you said that it won't cause the problem mentioned on the ticket and that we're going to replace this extension soon, I would say it's ok to leave this as it is.